compiler/src/iree/compiler/Codegen/Utils/Utils.h

```diff
@@ -73,12 +73,18 @@ const char *getIreeArchNameForTargetTriple(llvm::Triple triple);
 /// Methods to get target information.
 bool isVMVXBackend(IREE::HAL::ExecutableTargetAttr targetAttr);
 
+/// Methods to get target information.
+bool isROCMBackend(IREE::HAL::ExecutableTargetAttr targetAttr);
+
 // Returns true if the ukernel with given `ukernelName` is enabled.
 // If `ukernelName` is empty (the default), returns true if any ukernel
 // is enabled at all.
 bool hasUkernel(IREE::HAL::ExecutableTargetAttr targetAttr,
                 StringRef ukernelName = "");
 
+// Returns true if the rocm archtecture target is supported for ukernels.
+bool hasUkernelSupportedRocmArch(IREE::HAL::ExecutableTargetAttr targetAttr);
```

Comment:
This is specific to Rocm and shouldnt live here.

Reply:
Moved it to GPUUtils.h and make the API more general to also support cuda in the future.